src/features/post/schema.ts

```diff
@@ -0,0 +1,11 @@
+import * as z from "zod";
+
+export const post = z.object({
+  title: z.string().min(1, {message: "Le titre du post ne peut pas être vide"}),
+  category: z.string().min(1, {message: "Veuillez sélectionner une catégorie"}),
+  content: z
+    .string()
+    .min(1, {message: "Le contenu du post ne peut pas être vide"}),
+});
+
+export type PostPayload = z.infer<typeof post>;
```

Comment:
CreatePost -- use this type in useForm instead of the generated ts-client model,
your may perform transformation after validation